scripts/security-checker.mjs

```diff
@@ -69,29 +74,40 @@ class SecurityChecker {
 
   createAlertDictionary (existedIssues) {
       return existedIssues.reduce((res, issue) => {
-          const [, repo] = issue.body.match(/Repository:\s*`(.*)`/);
-          const [, url, type, number] = issue.body.match(/Link:\s*(https:.*\/(dependabot|code-scanning)\/(\d+))/);
+          const [, url, type] = issue.body.match(/(https:.*\/(dependabot|code-scanning)\/(\d+))/);
 
-          if (!url || repo !== this.context.repo)
+          if (!url)
               return res;
 
-          res[url] = { issue, number, type };
+          if (type === ALERT_TYPES.dependabot) {
+              const [, cveId] = issue.body.match(/CVE ID:\s*`(.*)`/);
+              const [, ghsaId] = issue.body.match(/GHSA ID:\s*`(.*)`/);
+
+              res.set(issue.title, { issue, type, cveId, ghsaId });
+          } 
+          else 
+              res.set(issue.title, { issue, type })
+
 
           return res;
-      }, {});
+      }, new Map());
   }
 
   async closeSpoiledIssues () {
-      for (const key in this.alertDictionary) {
-          const alert = this.alertDictionary[key];
+      for (const alert of this.alertDictionary.values()) {
 
           if (alert.type === ALERT_TYPES.dependabot) {
-              const isAlertOpened = await this.isDependabotAlertOpened(alert.number);
+              const matchAlertInIssue = alert.issue.body.match(new RegExp(`\`${this.context.repo}\` - https:.*/dependabot/(\\d+)`));
```

Comment:
1. matchAlertInIssue - is not an acceptable name for a variable because it starts with the verb
2. Let's create the regular expression outside the loop to optimize processing